metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/model/PartitionInfo.java

```diff
@@ -53,9 +58,17 @@ private PartitionInfo(
         final QualifiedName name,
         final AuditInfo auditInfo,
         final Map<String, String> metadata,
-        final StorageInfo serde
+        final StorageInfo serde,
+        final ObjectNode dataMetrics
     ) {
         super(name, auditInfo, metadata);
         this.serde = serde;
+        this.dataMetrics = dataMetrics;
+    }
+
+    @Override
+    public int compareTo(@NonNull @Nonnull final Object o) {
```

Comment:
We can add this to the BaseInfo.

Reply:
Can't move in BaseInfo since we need to cast the class to get Name. I will change the compareTo as an outside comparator.